app/views/subscriptions/index.html.erb

```diff
@@ -5,7 +5,7 @@
 
 <% @subscriptions.each do |sub| %>
```

Comment:
I would change this line to do what you want to achieve:

```ruby
<% @subscriptions.order(created_at: :asc).each_with_index do |sub, index| %>
  <details data-sub-id="<%= sub.id %>">
    <summary><%= sub.summary(index + 1) %></summary>
    <p>Subscription to <%= phrase_for sub.type, sub.qualifier %>, emailed every <%= pluralize(sub.frequency, 'day') 
```

(Or similar)

Now index+1 will be the nth subscription and can be used in the view, and you wouldn't need to do the query.

Reply:
Applied your other suggestion as per our discussion in Discord - adjusted it a little to make sure the content of the subscription is accessible in the loop, but it's still pretty much a `flat_map` -> `each_with_index` -> `map` -> `sort_by` -> `each` (xD)  chain.